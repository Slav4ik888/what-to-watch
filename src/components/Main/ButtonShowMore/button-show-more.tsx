import * as React from 'react';


type Props = {
  onShowMore: () => void,
  showButton: boolean,
};


const ButtonShowMore: React.FC<Props> = ({showButton, onShowMore}) => {
  if (!showButton) {
    return null;
  }

  const handleShowMore = () => {
    onShowMore();
  };

  return (
    <div
      className="catalog__more"
      onClick={handleShowMore}
    >
      <button className="catalog__button" type="button">Show more</button>
    </div>
  )
};

export default ButtonShowMore;
