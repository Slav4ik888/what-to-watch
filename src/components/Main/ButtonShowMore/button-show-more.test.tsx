import * as React from 'react';
import * as renderer from 'react-test-renderer';

import ButtonShowMore from './button-show-more';


describe(`Snapshot <ButtonShowMore />`, () => {
  it(`Render <ButtonShowMore /> showButton={true}`, () => {
    const tree = renderer
      .create(
          <ButtonShowMore
            onShowMore={() => {}}
            showButton={true}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render <ButtonShowMore /> showButton={false}`, () => {
    const tree = renderer
      .create(
          <ButtonShowMore
            onShowMore={() => {}}
            showButton={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
