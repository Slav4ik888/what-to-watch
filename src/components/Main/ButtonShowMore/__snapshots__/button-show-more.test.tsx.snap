// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot <ButtonShowMore /> Render <ButtonShowMore /> showButton={false} 1`] = `null`;

exports[`Snapshot <ButtonShowMore /> Render <ButtonShowMore /> showButton={true} 1`] = `
<div
  className="catalog__more"
  onClick={[Function]}
>
  <button
    className="catalog__button"
    type="button"
  >
    Show more
  </button>
</div>
`;
